server/datastore/mysql/microsoft_mdm.go

```diff
@@ -254,31 +254,32 @@ WHERE
 	return &res, nil
 }
 
-func (ds *Datastore) GetMDMWindowsBitLockerStatus(ctx context.Context, host *fleet.Host) (*fleet.DiskEncryptionStatus, error) {
+func (ds *Datastore) GetMDMWindowsBitLockerStatus(ctx context.Context, host *fleet.Host) (*fleet.HostMDMDiskEncryption, error) {
+	hde := &fleet.HostMDMDiskEncryption{}
 	if host == nil {
-		return nil, errors.New("host cannot be nil")
+		return hde, errors.New("host cannot be nil")
```

Comment:
Is there a specific reason why we return non-nil along with the errors? Usually when a call results in an error the other value is not checked, and is left `nil`.